content/blog/zephyr-tracing-and-profiling/index.md

```diff
@@ -0,0 +1,279 @@
+﻿---
+title: "Introduction to Zephyr OS Tracing and Profiling"
+date: 2024-10-25T06:54:17+08:00
+showAuthor: false
+featureAsset: "featured-gauge.webp"
+authors:
+  - "raffael-rostagno"
+tags: ["ESP32", "Tracing",  "Profiling","Zephyr"]
+---
+
+## Overview
+
+Embedded systems can be quite complex, depending on the architecture, application size and nature. Even for an experienced embedded developer, understanding the interplay of threads, interrupts and many processes that run in a large application can be challenging in a situation where debugging, evaluating hardware adequacy or optimizing application code is necessary. We humans are (mostly) visual by nature, and having the means to *visualize* what is happening in a given system can really open up possibilities. Without the right tools we are often in the dark, quite literally.
+
+One category of tools interesting to learn about is Tracing and Profiling. *Tracing* means to collect software execution data - the sequence and timing of execution of functions, threads, ISRs - and log it for posterior analysis, without interrupting or interfering (at best) in the application execution. *Profiling* aims to measure performance metrics such as execution time, CPU usage and dynamic memory usage. With such tools it is possible to evaluate how well an embedded system is performing, understand if the hardware used is adequate for the application being run and debug eventual problems.
+
+In this article we are going to do a brief introduction and show some basic examples of these resources as implemented in Zephyr RTOS. Later we’ll take a look at some of the features offered by Percepio’s Tracealyzer, which is a third party solution that extends some of Zephyr's *tracing and profiling* native functionalities and offers a front-end tool to analyze the data.
+
+
+## Zephyr’s tracing tool
+
+In order to show how Zephyr’s native tracing system works, we’ll use the sample below and an ESP32-C6 to run it on a physical device:

```

Comment:
I would suggest avoid using `will` in general:
`we’ll use the sample below and an ESP32-C6 to run it on a physical device:`
to
`ESP32-C6 board target is used with the following sample code:`